### PR TITLE
Fixed Error in Comment

### PR DESCRIPTION
**Pull request by mahony0**
Http library throws error if you use setOption key as string like **$http->setOption('CURLOPT_RETURNTRANSFER', true);**

> "curl_setopt_array(): Array keys must be CURLOPT constants or equivalent integer values" on line 261 of .../vendor/october/rain/src/Network/Http.php"

Both README and Http.php comments fixed by changing array key to the CURLOPT constant.

### CODE REVIEW

**Comment by bennothommo** [comment]
@mahony0 In addition to your docs changes, would it be worth converting a string to a `CURLOPT_` constant by using `defined()` to see if the constant exists, and use `constant()` to extract the value, if a string key is given?

**Comment by mahony0** [comment]
Sure, it seems pretty logical. I'll update the pull request

**Comment by mahony0** [comment]
Sent PR. Also created an issue related to this library on: https://github.com/octobercms/october/issues/4624

File: src/Network/Http.php [hunk 1]
```diff
@@ -520,7 +520,22 @@ public function setOption($option, $value = null)
...
+            $optionKey = constant($option);
+            $this->requestOptions[$optionKey] = $value;
+        } elseif (is_int($option)) {
+            $constants = get_defined_constants(true);
+            $curlOptConstants = preg_grep('/^CURLOPT_/', array_flip($constants['curl']));
+
+            if (isset($curlOptConstants[$option])) {
+                $this->requestOptions[$option] = $value;
+            } else {
+                throw new \Exception('Array keys must be CURLOPT constants or equivalent integer values');
```
Comment (bennothommo) [inline src/Network/Http.php]: @mahony0 Would you mind changing the Exception type to `October\Rain\Exception\ApplicationException`, to reflect that it is an implementation error by the developer?

Comment (bennothommo) [reply src/Network/Http.php]: In addition, would you also mind changing the message to the following?
```suggestion
                throw new \Exception('$option parameter must be a CURLOPT constant or equivalent integer');
```

File: src/Network/Http.php [hunk 2]
```diff
@@ -520,7 +520,22 @@ public function setOption($option, $value = null)
...
+            }
+        } else {
+            throw new \Exception('Array keys must be CURLOPT constants or equivalent integer values');
```
Comment (bennothommo) [inline src/Network/Http.php]: @mahony0 Same as above.

**Comment by LukeTowers** [comment]
@bennothommo could you add some tests for this?

**Pull request merged**